alphacsc/_encoder.py

```diff
@@ -122,9 +122,9 @@ def get_z_encoder_for(
         assert feasible_evaluation is False, (
             "DiCoDiLe requires feasible_evaluation=False."
         )
-        assert uv_constraint == 'auto',  (
-            "DiCoDiLe requires uv_constraint=auto."
-        )
+        # assert uv_constraint == 'auto',  (
+        #     "DiCoDiLe requires uv_constraint=auto."
+        # ) #XXX
```

Comment:
shall we remove this?

Reply:
Sure!

Regarding running time, the fit time was 2670.7s on my laptop (it seems to be the lion's share of the total running time)

Timing it on my machine gave

```
real	48m3,636s
```

Reply:
that's quite long to be honest. dicodile is supposed to be faster that alphacsc. Maybe it's due to greedy vs non greedy.

in any case this can be improved later as the results already look good !

Reply:
Yes, it would be interesting to see how this affects running time (5-way split in the original example vs no split here). I did not yet time the original example on my laptop but on the CI machine it's around 70 minutes I think?